lib/taskana-cdi/src/test/java/pro/taskana/common/internal/TaskanaProducersTest.java

```diff
@@ -1,43 +1,71 @@
 package pro.taskana.common.internal;
 
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.Statement;
-import javax.ws.rs.client.Client;
-import javax.ws.rs.client.ClientBuilder;
+import java.time.Duration;
+import org.apache.commons.io.FileUtils;
 import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.arquillian.junit.Arquillian;
 import org.jboss.shrinkwrap.api.Archive;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.FixMethodOrder;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
 import org.wildfly.swarm.undertow.WARArchive;
 
 @RunWith(Arquillian.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class TaskanaProducersTest {
 
+  private static final HttpClient HTTP_CLIENT =
+      HttpClient.newBuilder()
+          .version(HttpClient.Version.HTTP_1_1)
+          .connectTimeout(Duration.ofSeconds(10))
+          .build();
+
   @Deployment(testable = false)
   public static Archive<?> createDeployment() throws Exception {
     WARArchive deployment = ShrinkWrap.create(WARArchive.class);
     deployment.addPackage("pro.taskana");
     deployment.addClass(TaskanaProducers.class);
     deployment.addAllDependencies();
-    deployment.addDependency("org.mybatis:mybatis:3.4.2");
-    deployment.addDependency("org.mybatis:mybatis-cdi:1.0.0");
-    deployment.addDependency("pro.taskana:taskana-core");
     deployment.addAsResource("META-INF/beans.xml");
     deployment.addAsResource("taskana.properties");
     deployment.addAsResource("project-defaults.yml");
     return deployment;
   }
 
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    // Delete Taskana folder if exists
+    Path taskanaH2Data = Path.of(System.getProperty("user.home"), "taskana-h2-data");
+    if (Files.exists(taskanaH2Data)) {
+      FileUtils.deleteDirectory(taskanaH2Data.toFile());
+    }
+  }
+
   @Test
   public void testCommit() throws Exception {
 
-    Client client = ClientBuilder.newClient();
-    client.target("http://127.0.0.1:8090/rest/test").request().get();
+    HttpRequest getRequest =
+        HttpRequest.newBuilder(new URI("http://127.0.0.1:8090/rest/test")).GET().build();
+
+    HttpResponse<String> response =
+        HTTP_CLIENT.send(getRequest, HttpResponse.BodyHandlers.ofString());
+
+    // Assert.assertEquals(200, response.statusCode());
```

Comment:
Isn't this the crucial part of the test? Is it simply not working or why is it commented out?

Reply:
Hello @holgerhagen ,

yes it is, but it wasn't there before. :-( Currently the Test is not working because of missing configuration.

```
TaskanaException [errorCode=ErrorCode [key=DOMAIN_NOT_FOUND, messageVariables={domain=cdiDomain}], message=Domain 'cdiDomain' does not exist in the configuration]
```

When I'm back I will have a look on it together with @mustaphazorgati .

We will fix it

Reply:
Hello @holgerhagen , @mustaphazorgati 

I was able to fix it

please review my changes carefully